<launch>
    <arg name="rviz" default="true" />
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find localizer)/conf/run.conf" />
    <node pkg="voxel_mapping" type="voxel_mapping_node" name="voxel_mapping" output="screen">
        <param name="map_frame" value="odom" />
        <param name="scan_topic" value="/laser/scan" />
        <param name="body_frame" value="scan" />
        <param name="resolution" value="0.1" />
        <param name="plane_threshold" value="0.05" />
        <param name="max_iteration" value="10" />
        <param name="update_thresh" value="5.0" />
        <param name="max_update_thresh" value="2000.0" />
        <param name="ignore_range" value="0.1" />
        <param name="max_voxel_number" value="100000" />
    </node>
    <node pkg="localizer" type="localizer_node" name="localizer" output="screen">
        <param name="map_pcd_path" value="$(env HOME)/temp/sim2d/sim.pcd" />
        <param name="map_frame" value="map" />
        <param name="odom_frame" value="odom" />
        <param name="body_frame" value="scan" />
        <param name="point_topic" value="/voxel_mapping/points" />
        <param name="odom_topic" value="/voxel_mapping/odom" />
        <param name="localize_hz" value="0.5" />
        <param name="max_acc_num" value="5" />
        <param name="scan_resolution" value="0.01" />
        <param name="map_resolution" value="0.05" />
        <param name="k_near" value="10" />
        <param name="max_iteration" value="10" />
        <param name="max_search_distance" value="1.0" />
        <param name="plane_thresh" value="0.1" />
        <param name="score_thresh" value="0.05" />
    </node>

    <group if="$(arg rviz)">
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find localizer)/rviz/default.rviz" />
    </group>
</launch>